packages/room/peer/peer.js

```diff
@@ -210,34 +216,126 @@ export const createPeer = ({ api, createStream, event, streams, config }) => {
 
     /**
      * Turn on the local camera
+     * @param {MediaStreamTrack} [newTrack] sender video track
      */
-    turnOnCamera = () => {
-      if (!this._peerConnection) return
-      this._setTrackEnabled(this._peerConnection, 'video', true)
+    turnOnCamera = async (newTrack) => {
+      const localStream = this.getAllStreams().find((stream) => {
+        return stream.origin === 'local' && stream.source === 'media'
+      })
+
+      if (!localStream || !(localStream.mediaStream instanceof MediaStream)) {
+        throw new Error(
+          'Add local media stream with addStream() before calling this method'
+        )
+      }
+
+      const videoTrack = localStream.mediaStream.getVideoTracks()[0]
+
+      if (newTrack) {
+        if (videoTrack) {
+          localStream.replaceTrack(newTrack)
+          await this.replaceTrack(newTrack)
+          return
+        }
+
+        this._addVideoTrack(newTrack, localStream)
+        await this.negotiate()
+        return
+      }
+
+      if (videoTrack) {
+        this._setTrackEnabled(videoTrack, true)
+      }
```

Comment:
How about turn on camera without passing new track but previously was turnOff with stop? I don't see an error handler for that case here.

Reply:
Mas @tyohan  I have added some checkings that throw some errors on [1584b84](https://github.com/inlivedev/inlive-js-sdk/pull/109/commits/1584b84dddd68f347811827a2efe471106ef8ecd). Including turn on camera without passing new track but previously was turnedOff with stop will make the [track readyState](https://developer.mozilla.org/en-US/docs/Web/API/MediaStreamTrack/readyState) becomes `ended` and SDK should throw an error that display a message to use `turnOnCamera(newTrack)` instead.